File: 07_09_22/Remove_One_Element_to_Make_the_Array_Strictly_Increasing.py
class Solution:
    def canBeIncreasing(self, n: List[int]) -> bool:
        f=0
        for i in range(1,len(n)):
            if(n[i-1]>=n[i]):
                if(f):
                    return False
                f=1
                if(i>1 and (n[i-2]>=n[i])):
                    n[i]=n[i-1]
        return True

File: 07_09_22/Remove_One_Element_to_Make_the_Array_Strictly_Increasing.cpp
class Solution {
public:
    bool canBeIncreasing(vector<int>& n) {
        int f=0;
        for(int i=1;i<n.size();i++)
        {
            if(n[i-1]>=n[i])
            {
                if(f)
                {
                    return false;
                }
                f=1;
                if(i>1 && (n[i-2]>=n[i]))
                {
                    n[i]=n[i-1];
                }
                    
            }
        }
        return true;
        
    }
};